### ISSUE TITLE
Timed out mirrors produce corrupted data

### ISSUE THREAD

**Issue by danilaml** [JKSH/QtSdkRepoChooser#7]
For some reason the `master.qt.io` mirror (if I understood correctly - it's the default repo) almost always times out for me. For this reason the Qt Maintenance tool is always unusable for me as it is always times out on some of the metadata files. I tried to use this tool to change the repo and I saw this mirror in the list. I tried to use that mirror but each metadata download failed with connected timed out. As a result, in the repo folder there were a bunch of corrupted zip archives and QT Maintenance  Tool complained that checksums didn't match.

Btw, I'm unable to select different mirror until I restart the program, not sure if it's intended.

**Comment by JKSH** [comment]
Yep, reproduced 10 minutes ago: When I try to pick `master.qt.io` as the mirror, some parts timed out so they don't get downloaded, but the Qt SDK Repo Chooser doesn't realize this acts as if everything downloaded correctly.

Today was the first time I've encountered a timeout in this tool though (and I've used `master.qt.io` before). @danilaml, has this happened to you before?

I'm guessing that master server is getting hammered right now because [Qt 5.9.2](http://blog.qt.io/blog/2017/10/06/qt-5-9-2-released/) and [Qt Creator 4.4.1](http://blog.qt.io/blog/2017/10/06/qt-creator-4-4-1-released/) have just been released.
.

**Comment by danilaml** [comment]
>has this happened to you before?

@JKSH not from the use of this tool, that I can remember, but I used it only once and it didn't help/work at the time irc, but I didn't have this specific problem for sure.
